File: c#/program4.cs
using System;

class NumberX
{
    public int No;

    public NumberX(int no)
    {
        this.No = no;
    }

    public void DisplayFactors()
    {
        for(int i = 1; i <= (No/2); i++)
        {
            if((No % i) == 0)
            {
                Console.WriteLine(i+"\t");
            }
        }
        Console.WriteLine();
    }
}

class program4
{
    public static void Main(string[] args)
    {
        int value = 0;

        Console.WriteLine("Enter number : ");
        value = int.Parse(Console.ReadLine());

        NumberX obj = new NumberX(value);
        obj.DisplayFactors();
    }
}

File: scala/program4.scala
import scala.io.StdIn.{readInt}

class Factors(var num : Int)
{
    def DisplayFactors() = {
        var fact = for(i <- 1 to (num) if num % i == 0) yield i
        fact;
    }
}

object Demo extends App 
{
    println("Enter number :")
    var num : Int  = readInt();

    var obj = new Factors(num)
    var res = obj.DisplayFactors()
    println(res);
}